Removed old links and notes related to class components

niksseif opened the pull request:
This PR addressed issue https://github.com/turingschool/front-end-curriculum/issues/637

### Changes Requested in Issue:

The current prework has multiple links to the old react docs, and examples showing class-based components
These links should be updated to either more up to date alternatives, or removed entirely if they are no longer relevant
Confer with the current M3 team as to any additional resources that should be added or removed

### Changes Made:

- [x] Removed all the old links that was linked to the old React Docs
- [x] All links have been updated, but I've left a few links out to consult with the M3 team in case they are no longer relevant. Please refer to the question section for details.
- [ ] Confer with the current M3 team as to any additional resources that should be added or removed

### Question:

I'm unsure whether to keep these three links in our intermission work, as they are dated and I don't know if they're still helpful or if students even refer to them.See below 

- Take a look at some extra [Testing with Fun Fun Function](https://www.youtube.com/playlist?list=PL0zVEGEvSaeF_zoW9o66wa_UCNE3a7BEr) (This is a playlist of 7 YouTube videos on testing) - this is about unit testing
- [Shop Talk with Tom Dale](http://shoptalkshow.com/episodes/147-tom-dale/) (This one is a bit older but still a good introduction for thinking about why client-side applications are important and what frameworks bring to the table.) - This one is dated too but I think it's still relevant. 
-  [Wes Bos JavaScript30](https://javascript30.com) (Maybe pick 5 or 10 exercises - try to challenge yourself) - This resource is dated but contains some valuable JavaScript exercises. Instead of including this link, perhaps we could provide a link to CodeWars for more up-to-date and engaging practice opportunities?

CassandraGoose reviewed (approved):
I'm not a Mod 3 instructor but I figured I'd throw a review on anyways!

Looks great, Nik!  I 

I agree that the Testing with Fun Fun Function video may not be incredibly helpful anymore. I actually think it’s a little confusing to introduce E2E testing and then include a video on Unit testing right after.  My vote is if we want extra testing resources, add something related to Cypress or E2E testing in general instead. 

I’m on the fence about the Wes Bos JS30 challenges. A lot of them are actually DOM-related things. I’d rather get their brain thinking in React and Interviewing, so I dig your idea about adding in code wars instead, since that’s a nice building block towards the code challenges they may run into on the job hunt. 

🎉

niksseif commented:
Thanks, Cass! I have changed those links here 5e0dfc07dda6efba0597f23e0d85fa140f7c3270

hfaerber commented:
I'm glad to see we've removed Testing with Fun Fun Function and replaced the Wes Bos JS30 with Code Wars.

I wonder if we can find a similarly valuable but more recent video or podcast to replace Shop Talk with Tom Dale.  2014 feels like ancient history.  I think we can leave it for now and I'll start looking for other options that we could replace it with down the road.

hfaerber commented on prep-for-module-3-frontend.markdown:
```diff
@@ -10,11 +10,10 @@
-* [Tyler McGinnis's React Tutorial: A Comprehensive Guide to learning React.js in 2018](https://tylermcginnis.com/reactjs-tutorial-a-comprehensive-guide-to-building-apps-with-react/)
-* [React tutorial from the docs](https://reactjs.org/tutorial/tutorial.html)
-* [Original Scrimba React Course](https://scrimba.com/playlist/p7P5Hd)
+* [Tyler McGinnis's React Tutorial: A Comprehensive Guide to learning React.js in 2018](https://ui.dev/react-hooks)
```
This Tyler McGinnis one costs $149, I couldn't find any free trial or other free option.  I think we should remove it.

niksseif replied:
I removed it for now until we find a better option

hfaerber commented on prep-for-module-3-frontend.markdown:
```diff
@@ -10,11 +10,10 @@
-* [Original Scrimba React Course](https://scrimba.com/playlist/p7P5Hd)
+* [Tyler McGinnis's React Tutorial: A Comprehensive Guide to learning React.js in 2018](https://ui.dev/react-hooks)
+* [React tutorial from the docs](https://react.dev/learn/tutorial-tic-tac-toe)
+* [Original Scrimba React Course](https://scrimba.com/learn/learnreact) - In M3 the initial focus will be on learning functional components and the utilization of hooks.
```
This Scrimba new course is 12 hours long.  I think its good that its linked but we'll need to give students guidance on what to skip/focus on.   The old course was 6 hours - would it be better to let them keep using that one and have them skip the Class-based parts and really focus on hooks at the end?

niksseif replied:
Added that to the list

hfaerber commented on prep-for-module-3-frontend.markdown:
```diff
@@ -66,12 +68,13 @@ Create a GitHub gist to answer these questions in as much detail as possible. Im
 
-_Note:_ As you do your research, stay away from React articles that talk exclusively about "hooks". We _will_ be learning hooks in Mod 3 but they are not a requirement for success in the module.
-
+_Note:_ As you do your research, be aware that if you come across tutorials discussing class components, they may be related to older versions of React. We used to rely on class components before hooks were introduced. We _will_ be learning hooks in Mod 3, and they are now the more modern and recommended approach.  
```
I'd love to see us add a note here that its still very possible that they will run into class-based React components in legacy code on the job.  I don't want them to think that class-based components have been completed phased out.

hfaerber commented on prep-for-module-3-frontend.markdown:
```diff
@@ -83,13 +86,13 @@ Please update your LinkedIn profiles with...
 
-- Take a deeper dive into JSX with the [React docs](https://reactjs.org/docs/introducing-jsx.html) and this [tutorial](https://flaviocopes.com/jsx/)
-- Take a look at some extra [JSX gotchas (courtesy of tyler McGinnis)](https://ui.dev/jsx/)- [Testing with Fun Fun Function](https://www.youtube.com/playlist?list=PL0zVEGEvSaeF_zoW9o66wa_UCNE3a7BEr) (This is a playlist of 7 YouTube videos on testing)
+- Take a deeper dive into JSX with the [React docs](https://react.dev/learn/javascript-in-jsx-with-curly-braces) and this [tutorial](https://flaviocopes.com/jsx/)
```
Can we change "tutorial" to "article"?  I was checking it out and it doesn't actually include a tutorial.

hfaerber reviewed (changes requested):
Thanks for updating this Nik!  I requested a few changes and we might need to spend some time figuring out what we want to do for the Scrimba course.  I remember the 6 hour course been tough to get all the way through over intermission and am really worried about the length of the new 12 hours course

niksseif commented:
@hfaerber I have added the scrimba original lesson, removed the $179 course and add a note about class component in legacy code. Let me know if there anything else you would like to change. Thanks
